challenges/2021-08-31-zero-point/solutions/rust/ericwburden/ericwburden.rs

```diff
@@ -0,0 +1,249 @@
+//------------------------------------------------------------------------------------- 
+// Here we set up the LinkedList struct and parts
+//------------------------------------------------------------------------------------- 
+#[derive(Debug)]
+pub struct Node<T> {
+    value: T,
+    next: Link<T>,
+}
+
+type Link<T> = Option<Box<Node<T>>>;
+
+#[derive(Debug)]
+pub struct LinkedList<T> {
+    head: Link<T>,
+}
+
+impl<T> LinkedList<T> {
+    pub fn new() -> Self {
+        LinkedList { head: None }
+    }
+
+    pub fn from(values: Vec<T>) -> Self {
+        let mut out = Self::new();
+        for value in values {
+            out.push_value(value);
+        }
+        out
```

Comment:
I've seen this a few times, is this a way to "return" the object of a "class"? using these words loosely.

Reply:
Rust (like R and Julia, actually) doesn't make you use a return statement. Instead, it returns the last un-terminated expression (in this case the `out` vector). It's equivalent to `return out;`.